scalyr_agent/scalyr_monitor.py

```diff
@@ -121,6 +121,13 @@ def __init__(
         # The logger instance that this monitor should use to report all information and metric values.
         self._logger = logger
         self.monitor_name = monitor_config["module"]
+
+        # Flag that indicates that the agent can not run without this monitor and
+        # has to fail if monitor fails.
+        # The default value (False) of this config option has to be already set for all monitors,
+        # but also add it here for extra safety.
+        self.stop_agent_on_failure = monitor_config.get("stop_agent_on_failure", False)
```

Comment:
I would actually still prefer this to be a getter method since it gives us more flexibility in overriding that logic in sub classes (I know that technically we can also use ``@property``, but ``get_stop_agent_on_failure()`` is a bit more explicit.

Reply:
EDIT: I had a look and it appears existing base ``ScalyrMonitor`` uses both patterns - ``get_()`` method and ``@property`` decorator.

So as far as consistency goes, both methods would probably work, but getter method is probably preferred since it's much more widely uses than ``@property`` (e.g. ``get_user_agent_fragment()``, ``get_calculate_rate_metric_names()``, etc.).